src/main/java/doke/TaskList.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    public void searchString(String str, Ui ui) {
+        int len = taskList.size();
+        int num = 0;
+        String message = "_________________________ \n";
+        int i = 0;
+        message = "_________________________ \n";
+        while (i < len) {
+            Task task = taskList.get(i);
+            String desc = task.getDesc().toLowerCase();
+            String taskType = task.getType();
+            if (desc.indexOf(str.toLowerCase()) != -1) {
+                if (num == 0) {
+                    message += "Here's what we found: \n";
+                }
+                message += (i + 1) + "." + taskList.get(i).toString() + "\n";
+                num += 1;
+            } else if ((taskType.equals("E") || taskType.equals("D"))
+                    && task.getTime().format(
+                            DateTimeFormatter.ofPattern("dd-MMM-yyyy")).indexOf(str) != -1) {
+                if (num == 0) {
+                    message += "Here's what we found: \n";
+                }
+                message += (i + 1) + "." + taskList.get(i).toString() + "\n";
+                num += 1;
+            }
+            i++;
+        }
+        if (num == 0) {
+            message += "Sorry, we found nothing. \n";
+        }
+        message += "_________________________ \n";
+        ui.printOut(message);
+    }
+
+    /**
+     * Sends order to ui to print out the whole content of the ArrayList.
+     *
+     * @param ui
+     */
+    public void listOut(Ui ui) {
+        String message;
+        if (this.taskList.isEmpty()) {
+            message = "_________________________ \n" + "You have no task! \n"
+                    + "________________________";
+            ui.printOut(message);
+            return;
+        }
+        int len = this.taskList.size();
+        int i = 0;
+        message = "_________________________ \n";
+        while (i < len) {
+            message += (i + 1) + "." + this.taskList.get(i).toString() + "\n";
+            i++;
+        }
+        message += "_________________________ \n";
+        ui.printOut(message);
+    }
+
+    /**
+     * Deletes the ith task from the ArrayList.
+     *
+     * @param i the position + 1 of the task to delete
+     */
+    public void deleteTask(int i) {
+        this.taskList.remove(i - 1);
+    }
+
+    /**
+     * Adds a task to the ArrayList.
+     *
+     * @param task the task to be added.
+     */
+    public void addTask(Task task) {
+        this.taskList.add(task);
+    }
+
+    /**
+     * Returns the Task at the position i-1 in the ArrayList.
+     *
+     * @param i the position + 1 of the task wanted.
+     * @return the task desired.
+     */
+    public Task getTask(int i) {
+        return taskList.get(i - 1);
+    }
+
+    /**
+     * Returns a copy of the ArrayList.
+     *
+     * @return a copy of the ArrayList.
+     */
+    public ArrayList<Task> getList() {
+        ArrayList<Task> temp = this.taskList;
+        return temp;
+    }
```

Comment:
There seems to be an error in this function, it mentions that it would like to return a copy of the ArrayList, but the current implementaiton will result in aliasing the list. Perhaps consider cloning and casting the result?

```suggestion
     * Returns a copy of the ArrayList.
     *
     * @return a copy of the ArrayList.
     */
    public ArrayList<Task> getList() {
        ArrayList<Task> temp = (ArrayList<Task>)this.taskList.clone();
        return temp;
    }
```